### ISSUE TITLE
Implements 'create checklist' button on card

### ISSUE THREAD

**Issue by gabrielcamurcab** [nextcloud/deck#5487]
**Reason:**

Deck is a great KanBan app on Nextcloud, but lacks a quick way to add checklists. This issue proposes adding a button to the card for instant checklist creation, making the process more efficient and straightforward.

**Usability:**

- Enhanced Efficiency: A visible button on the card allows users to create checklists quickly without additional navigation.
- Intuitive Experience: Simplifies the workflow, making checklist addition an intuitive task.

**Functionality Suggestion:**

- Addition of a visible button on the card to create a checklist directly, without the need to access additional menus.
- Ability to name and add items to the checklist during the initial creation.

**Comment by juliushaertl** [comment]
Thanks for your suggestion. We already have a dedicated checklist item in the description editor so I'm wondering if you considered that or if there were any specific reasons this doesn't work for you:

<img width="981" alt="Screenshot 2024-01-18 at 08 59 46" src="https://github.com/nextcloud/deck/assets/3404133/bb9fb887-2cf2-4aaa-9525-acd0c110d842">

**Comment by apg1980** [comment]
as an addition to this topic, the latest deck app on nextcloud 28.0.1 did not show d´the editor for description if no older description is set, new cards cannot be editited because the edititor did not appear.

**Comment by juliushaertl** [comment]
@apg1980 Please open a separate issue with the full issue template filled out (especially logs and browser console errors)

**Comment by Ellipse0934** [comment]
> Addition of a visible button on the card to create a checklist directly, without the need to access additional menus

This clarifies a bit. And FWIW I agree with the idea. OP is suggesting that there be another button along with the ellipsis to create a tasklist.

This is a fair argument considering this is how people use kanban boards. I don't have a study to quote but I guess people first add cards, fill in due dates, tag it and add tasks. Having a flexible markdown and adding in links to other resources is great but maybe that can be behind the ellipsis. However, OP's solution may not be acceptable to many because tags, assignments and due dates are also quite important. Consider the following alternatives:

1) Clicking + to add a new card opens up the expanded view. This would be an improvement in my eyes even if I need to enable it in the settings.
2) Clicking + to add a new card brings up a slightly expanded view which shows the most used options.

![image](https://github.com/nextcloud/deck/assets/7466364/0ea4c3d9-a972-461a-a677-5807c2b2f122)

So, there is a textbox for name, and in a row underneath we can click to edit tags, assign people, set due date write a checklist.

In short, filling up the critical details as soon as we click new card button. Rather than creating and then re-navigating.

On a related note I originally found this more annoying because I didn't realize that clicking a card brings up the Card Details modal. I thought I will necessarily edit the card name so I had to go via the ellipsis which was too many clicks imo. My comment will sound nitpicky but I feel this will help user experience.

**Comment by toxpal** [comment]
> Thanks for your suggestion. We already have a dedicated checklist item in the description editor so I'm wondering if you considered that or if there were any specific reasons this doesn't work for you:
> <img alt="Screenshot 2024-01-18 at 08 59 46" width="981" src="https://private-user-images.githubusercontent.com/3404133/297666513-bb9fb887-2cf2-4aaa-9525-acd0c110d842.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTAxNzczMjEsIm5iZiI6MTcxMDE3NzAyMSwicGF0aCI6Ii8zNDA0MTMzLzI5NzY2NjUxMy1iYjlmYjg4Ny0yY2YyLTRhYWEtOTUyNS1hY2QwYzExMGQ4NDIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDMxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAzMTFUMTcxMDIxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9MDIxMWRjMjVlYzU5ZWYzNjI4N2MzZmQ3MDgzNGM2YTEzNjc4Mjk0YjYyNjY3MzA5NzcwMWM1N2E4MTIzMjkyOCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Zh4qENrURJtTtSO8CE2tMlq1cAV15tGRuun0ORMIiQ0">

Please excuse my ignorance, but how can I enable this markdown editor? Just installed the latest version of Deck on NC 28.0.3 and my description editor is very different - https://ibb.co/jRSFGGx

I use FireFox

**Comment by juliushaertl** [comment]
You need to have the text app enabled, is that the case on your instance?

**Comment by toxpal** [comment]
Thank you for the clarification, Text app wasn't enabled.

Just a suggestion: this should be explained in the documentation and/or readme because I'm sure there are many more users who install Deck without having Text app enabled, and the uninstall it because of "lack of features" (just because they have no idea another app needs to be enabled for Deck to offer advanced features).
